client/components/domains/register-domain-step/index.jsx

```diff
@@ -140,7 +141,35 @@ class RegisterDomainStep extends React.Component {
 		vendor: 'domainsbot',
 	};
 
-	state = this.getState();
+	constructor( props ) {
+		super( props );
+
+		resetSearchCount();
+
+		this._isMounted = false;
+		this.state = this.getState();
+		if ( props.initialState ) {
+			this.state = { ...props.initialState, railcarId: this.getNewRailcarId() };
+
+			if (
+				this.state.lastSurveyVertical &&
+				this.state.lastSurveyVertical !== props.surveyVertical
+			) {
+				this.state.loadingResults = true;
+
+				if ( props.includeWordPressDotCom || props.includeDotBlogSubdomain ) {
+					this.state.loadingSubdomainResults = true;
+				}
+
+				delete this.state.lastSurveyVertical;
+			}
+
+			if ( props.suggestion ) {
+				this.state.lastQuery = props.suggestion;
+				this.state.loadingResults = true;
```

Comment:
I think this should be `this.state.loadingResults = false;` right? If it's `true`, then the suggestions will never load from `/start/domains/?new=myquery`.

Reply:
It's a little more complicated than that - we should ensure that we have results to render before blindly setting `loadingResults` to `false`; 3e65fd8aee56eb4655672a642a21d058c9b0ee9a should take care of this.